spec/controllers/forgot_passwords_controller_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'spec_helper'
+
+describe ForgotPasswordsController do  
+  describe "POST create" do
+    after { ActionMailer::Base.deliveries.clear }
+
+    context "when the inputted email matches a user's email" do
+      let(:alice) { Fabricate(:user)}
+      before { post :create, email: alice.email }
+
+      it "creates a token for the user" do
+        expect(alice.reload.token).to be_present
+      end
+
+      it "sends an email to the user" do
+        expect(ActionMailer::Base.deliveries.last.to).to eq([alice.email])
+      end
+
+      it "sends an email with a url based on the newly created token" do
+        expect(ActionMailer::Base.deliveries.last.body).to include(reset_password_path(alice.reload.token))
+      end
+
+      it "redirects_to reset_password_path" do
+        expect(response).to redirect_to confirm_password_reset_path
+      end
+    end
+
+    context "when the inputted email does not match a user's email" do
+      before { post :create, email: "test@test.com" }
+
+      it "does not send an email" do
+        expect(ActionMailer::Base.deliveries).to be_empty
+      end
+
+      it "sets an error message" do
+        expect(flash[:danger]).to be_present
+      end
+
+      it "renders 'new' template" do
+        expect(response).to render_template 'new'
+      end
+    end
+  end
+
+  describe "GET edit" do
+    context "when the token is valid" do
+      let!(:alvin) { alvin = Fabricate(:user, token: 'hey') }
```

Comment:
There is no need for the `alvin =` here, it is a local variable which is immediately discarded.